src/libcore/mem.rs

```diff
@@ -1015,13 +1017,16 @@ impl<T> Deref for ManuallyDrop<T> {
 }
 
 #[stable(feature = "manually_drop", since = "1.20.0")]
-impl<T> DerefMut for ManuallyDrop<T> {
+impl<T: ?Sized> DerefMut for ManuallyDrop<T> {
     #[inline]
     fn deref_mut(&mut self) -> &mut Self::Target {
         &mut self.value
     }
 }
 
+#[unstable(feature = "coerce_unsized", issue = "27732")]
+impl<T: CoerceUnsized<U>, U> CoerceUnsized<ManuallyDrop<U>> for ManuallyDrop<T> {}
```

Comment:
Uh, no, this allows `ManuallyDrop<Rc<T>>` to coerce to `ManuallyDrop<Rc<dyn Trait>>` - do we *need* that?

Reply:
Is this change even useful without a `CoerceUnsized`? AFAIK all types that do `T: ?Sized` also have a `CoerceUnsized`, or is that not the case?

Reply:
In your test, the only needed `CoercedUnsized` impl is that for `Box`, pointers *to* `ManuallyDrop` can be coerced just because it's a `?Sized` struct (you can test this with any such structs, there's no opt-in there).

Reply:
Okay, done.